File: Max_Count_Postive_Negative/max.cs
using System;
using System.Collection.Generic;

public class Solution
{
    public int MaximumCount(int[] nums)
    {
        int lower = binarySearch(nums)[0];
        int upper = binarySearch(nums)[1];

        int negative = 0,
            positive = 0;

        for (int i = 0; i < lower; i++)
        {
            if (nums[i] < 0)
            {
                negative++;
            }
        }

        for (int i = upper; i < nums.Length; i++)
        {
            if (i >= 0 && nums[i] > 0)
            {
                positive++;
            }
        }

        return Math.Max(negative, positive);
    }

    private int[] binarySearch(int[] nums)
    {
        int i = 0,
            j = nums.Length - 1;

        while (i <= j)
        {
            int mid = i + (j - i) / 2;

            if (nums[mid] < 0)
            {
                i = mid + 1;
            }
            else
            {
                j = mid - 1;
            }
        }

        return new int[] { i, j };
    }
}

File: Max_Count_Postive_Negative/max.js
/**
 * @param {number[]} nums
 * @returns {number}
 */
var maximumCount = function (nums) {
	const { lower, upper } = binarySearch(nums);
	let [negative, positive] = [0, 0];

	for (let i = 0; i < lower; i++) {
		if (nums[i] < 0) {
			negative++;
		}
	}

	for (let j = upper; j < nums.length; j++) {
		if (j >= 0 && nums[j] > 0) {
			positive++;
		}
	}

	return Math.max(negative, positive);
};

/**
 * Binary search to find lower and upper
 * @param {number[]} numbers - The target number array
 * @typedef {object} boundInfo - Boundary data includes lower and upper bow
 * @property {number} lower - lower boundary
 * @property {number} upper - upper boundary
 * @returns {boundInfo} returns boundary info
 */
const binarySearch = (numbers) => {
	let [i, j] = [0, numbers.length - 1];

	while (i <= j) {
		let mid = Math.floor(i + (j - i) / 2);

		if (numbers[mid] < 0) {
			i = mid + 1;
		} else {
			j = mid - 1;
		}
	}

	return { lower: i, upper: j };
};
